fenix/app/src/main/java/org/mozilla/fenix/compose/SwitchWithLabel.kt

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * UI for a switch with label that can be on or off.
+ *
+ * @param checked Whether or not the switch is checked.
+ * @param onCheckedChange Invoked when Switch is being clicked, therefore the change of checked
+ * state is requested.
+ * @param label Text to be displayed next to the switch.
+ * @param modifier Modifier to be applied to the switch layout.
+ * @param enabled Whether the switch is enabled or grayed out.
+ */
+@Composable
+fun SwitchWithLabel(
+    checked: Boolean,
+    onCheckedChange: ((Boolean) -> Unit),
+    label: String,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Text(
+            text = label,
+            color = if (enabled) {
+                FirefoxTheme.colors.textPrimary
+            } else {
+                FirefoxTheme.colors.textDisabled
+            },
+            style = FirefoxTheme.typography.subtitle1,
+            modifier = Modifier.weight(1f),
+        )
+
+        Switch(
+            checked = checked,
+            onCheckedChange = onCheckedChange,
+            enabled = enabled,
+        )
+    }
+}
+
+/**
+ * UI for a switch that can be on or off.
+ *
+ * @param checked Whether or not the switch is checked.
+ * @param onCheckedChange Invoked when Switch is being clicked, therefore the change of checked
+ * state is requested.
+ * @param modifier Modifier to be applied to the switch layout.
+ * @param enabled Whether the switch is enabled or grayed out.
+ */
+@Composable
+private fun Switch(
+    checked: Boolean,
+    onCheckedChange: ((Boolean) -> Unit),
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    MaterialSwitch(
+        checked = checked,
+        onCheckedChange = onCheckedChange,
+        modifier = modifier,
+        enabled = enabled,
+        colors = SwitchDefaults.colors(
+            uncheckedThumbColor = FirefoxTheme.colors.formOff,
+            uncheckedTrackColor = FirefoxTheme.colors.formSurface,
+            checkedThumbColor = FirefoxTheme.colors.formOn,
+            checkedTrackColor = FirefoxTheme.colors.formSurface,
+            disabledUncheckedThumbColor = FirefoxTheme.colors.formOff
+                .copy(alpha = DISABLED_ALPHA)
+                .compositeOver(FirefoxTheme.colors.formSurface),
+            disabledUncheckedTrackColor = FirefoxTheme.colors.formSurface.copy(alpha = DISABLED_ALPHA),
+            disabledCheckedThumbColor = FirefoxTheme.colors.formOn
+                .copy(alpha = DISABLED_ALPHA)
+                .compositeOver(FirefoxTheme.colors.formSurface),
+            disabledCheckedTrackColor = FirefoxTheme.colors.formSurface.copy(alpha = DISABLED_ALPHA),
+        ),
+    )
+}
+
+@LightDarkPreview
+@Composable
+private fun SwitchWithLabelPreview() {
+    FirefoxTheme {
+        Column(
```

Comment:
time allowing, it'd be nice to have the Switches interactable in the preview here (i.e. able to have demo their toggles) and the disabled state could be validated.

Reply:
good idea, done!